crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1266,7 +1266,9 @@ impl<'db> Type<'db> {
     pub fn to_meta_type(&self, db: &'db dyn Db) -> Type<'db> {
         match self {
             Type::Never => Type::Never,
-            Type::Instance(InstanceType { class, .. }) => Type::ClassLiteral(*class),
+            Type::Instance(InstanceType { class, .. }) => {
+                Type::ClassLiteral(ClassLiteralType { class: *class })
```

Comment:
Just occurred to me that we should probably add a TODO comment here, because this is actually not quite right (predating this PR).

The correct meta type of an Instance type would be a `type[]` type that includes subclasses, not a class literal type.

Reply:
And worth noting that our choice to invest in improved naming of types is what brought this subtle issue to my attention!